Adds base trap class

mgodsy opened the pull request:
Adds trap bar function in trap class
Creates trap in playstate - can be removed, was just there for testing

Told you I'd be last min - Mark

mgodsy committed:
Adds base trap class
Adds trap bar function in trap class
Creates trap in playstate - can be removed, was just there for testing

SamBumgardner commented:
Don't forget to request a review from your reviewer person!

CheezitBandit commented:
I think I was supposed to review yours - I'll look at it this afternoon when I get the chance before class.

CheezitBandit commented on source/PlayState.hx:
```diff
@@ -2,17 +2,25 @@ package;
+	private var trap:Trap;
+	private var trap1:Trap;
+	private var trap2:Trap;
+	private var trap3:Trap;
```
Maybe rename these slightly? The variable names make it sound like there are four separate traps, instead of the four pieces of the bar.

CheezitBandit commented on source/Trap.hx:
```diff
@@ -0,0 +1,70 @@
+
+
+    //Builds flame bar - adds tweens (WIP)
+    public function barCreate(trap1:Trap, trap2:Trap, trap3:Trap, x:Float, y:Float)
```
You could use more comments describing what's going on in this class, in the style of the other comments above.

CheezitBandit commented on source/Trap.hx:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    //Overlap Resolution (WIP)
+    public function playerTrapResolve(player:Player, trap:Trap):Void
```
This could also use more documentation.

CheezitBandit commented on source/Trap.hx:
```diff
@@ -0,0 +1,70 @@
+    public function playerTrapResolve(player:Player, trap:Trap):Void
+    {
+        //player.takeHit()
+        player.kill();
```
OPTIONAL: Maybe the player should just take a hit or lose some health instead of being immediately killed by the bar? The red danger color makes it pretty obvious the trap isn't a good thing, but maybe the player should flash after one hit and lose health and then be killed after two hits.

mgodsy committed:
Cleaned up and automated alot of the Flame bar's creation and placement
Functions Added to Trap - buildTrap, getSubtrap, placeTrap
Commented the hell out of the code (Had alot more time this commit to sit down and explain everything)

CheezitBandit reviewed (approved):
Looks awesome. I think it was a good idea to make the bar smaller. It's positioned a little above the platform, but since it doesn't obey physics I don't think that matters. Looks great, thanks for adding the documentation!

SamBumgardner commented:
Feel free to merge after fixing any conflicts!

mgodsy committed:
Merge branch 'master' into Traps_(Mark)

# Conflicts:
#	source/PlayState.hx

merge conflicts resolved
